core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -953,6 +953,9 @@ private Object readResolve() {
     @Nonnull
     @Restricted(NoExternalUse.class)
     public InstallState getInstallState() {
+        if (installState == null || installState.name() == null) {
```

Comment:
Not sure why this is hackish? constructor is defined as: `InstallState(@Nonnull String name...`, only way for installState.name to be null is that it's corrupt, in which case we simply don't know what the state is (hence, `UNKNOWN`), and `proceedToNextStateFrom(UNKNOWN)` will be called.